### PR TITLE
Adding search function to xenit.github.io

### LINKED ISSUE #52: Add search function to docs page
The documents are growing fast and it's starting to be hard to get a quick overview of all the docs that we have.
To solve this add a **simple** search function to the page so we can find the docs quicker.

### PR DESCRIPTION
**Pull request by CalleB3**
Updates dependencies and adds docusaurus search function to our docs.

Resolves #52

### CODE REVIEW

**Review by bittrance — COMMENTED** [review]
Some questions I think the PR should address:
- Why did we pick this particular plugin? Which plugins were evaluated?
- Why do we need to upgrade React?
- Why do we need to add the svgr plugin, url-loader, file-loader as a production dependencies?

Testing out the search function does not give great result. Searching for "AKS" and "service principal" gives rather confusing hits. The plugin is focused on headings: I suspect we will need to go through the headings and make them more globally unique for the search experience to become decent.

File: .gitignore [hunk 1]
```diff
@@ -7,6 +7,7 @@
 # Generated files
 .docusaurus
 .cache-loader
+/src/theme/
```
[the remark this answers is not in the document]
Comment (ahockersten) [reply .gitignore:10]: I agree. I also do not get this one when testing, not sure how that happened when we were testing, but we should remove this.

**Comment by ahockersten** [comment]
@bittrance for me search results seem to include non-headings just fine though, or am I missing something?
image

**Comment by CalleB3** [comment]
@bittrance 
We have now looked at the other docusaurus search plugins, they are very similar. But we like this one the best.
Se Anders comment above regarding you other questions.

**Comment by bittrance** [comment]
@ahockersten I think your screenshot is rather confusing. Should I pick "Azure > Azure" or "Azure > Azure" to get what I want?

**Comment by NissesSenap** [comment]
What happened with this @CalleB3 @ahockersten ? 
For me something is better than nothing. If we have something that works today but isn't ideal I would rather get that. When we have time to make it better go for it.

**Comment by ahockersten** [comment]
I'm fine with merging this as-is, although think @CalleB3 "owns" it. LMK if you want help with how to resolve the merge conflicts though.

@bittrance I think #79 should not stop this from getting merged since it's already very close to being finished?

**Comment by bittrance** [comment]
@ahockersten Only in the sense that #79 implies that we may need to switch from Docusaurus altogether.

**Comment by CalleB3** [comment]
@NissesSenap @bittrance 
Conflicts fixed, please review for merging :)

**Pull request merged**